### ISSUE TITLE
Spec for column serialization (download)

### ISSUE THREAD

**Issue by alisman** [cBioPortal/cbioportal-frontend#169]
For the following columns, I need spec on how they should be serialized in export (csv or copy)

 Annotation
 Allele Freq
 Allele Freq (N)
 Copy #
 mRNA Expr.
 Cohort
 COSMIC

**Comment by ecerami** [comment]
@alisman:  sorry for the delay on this.  Actually, all the columns are pretty complicated.

There are two issues:

1.  If we have multiple samples, there is no way to do this simply, because under the hood, we have multiple genomic alterations that are merged at the protein level.  To preserve this would require enabling download of the underlying MAF file, which does not actually mirror the web UI.

2.  For annotations, e.g. from OncoKB or CIVIC, etc, there are multiple annotations, and not easy to serialize.  Plus, additional annotations will be added over time.

Since the copy/download is a also a new feature, I vote that we keep this as simple as possible, and only serialize columns that are the most important and do not result in errors.  My proposal is below.

Tumors: serialized as sample index values, e.g. 1,3, and 4
Gene:  yes, serialized as string
Protein Change:  yes, serialized as string
Annotation:  not serialized
Chr: not serialized
Start: not serialized
End: not serialized
Ref: not serialized
Var: not serialized
Status: not serialized
Validation: not serialized
Type: not serialized
Center: not serialized
Allele Freq: serialized as comma-separated list of floating point values, e.g. 0.2, 0.3
Variant Reads: not serialized
Ref Reads: not serialized
Allele Freq (N): not serialized
Variant Reads (N): not serialized
Ref Reads (N): not serialized
Copy #: yes, serialized as discrete values, e.g. -2, -1, 0, 1, 2
mRNA Expr.:  yes, serialized as floating values, e.g. .05
Cohort:  yes, serialized as floating values, e.g. 0.05
COSMIC: not serialized
Mutation Assessor: not serialized

We can revisit / add additional columns in the future, if users request them.

**Comment by alisman** [comment]
Hi Ethan, 
So, based on this list, the columns left to be serialized are: Allele Freq, Copy#, mRNA Expr and cohort.  Of these, I believe only Allele Freq is primary data, meaning for the others, we have issue of lazy loading.  I.e. for most results, we don't have the data yet to serialize.  
Retrieving all of the data in order to serialize it is perfectly possible, but a little more involved.  Let me know if we this is something we should consider.

**Comment by ecerami** [comment]
@alisman:  good point.  I vote to keep it even simpler then, e.g. only provide download for core fields.  My amended proposal:

Tumors: serialized as sample index values, e.g. 1,3, and 4
Gene: yes, serialized as string
Protein Change: yes, serialized as string
Annotation: not serialized
Chr: not serialized
Start: not serialized
End: not serialized
Ref: not serialized
Var: not serialized
Status: not serialized
Validation: not serialized
Type: not serialized
Center: not serialized
Allele Freq: serialized as comma-separated list of floating point values, e.g. 0.2, 0.3
Variant Reads: not serialized
Ref Reads: not serialized
Allele Freq (N): not serialized
Variant Reads (N): not serialized
Ref Reads (N): not serialized
Copy #: not serialized
mRNA Expr.: not serialized
Cohort: not serialized
COSMIC: not serialized
Mutation Assessor: not serialized
